src/shim/sdk/UserSDK.ts

```diff
@@ -34,30 +34,44 @@ export function rotateMasterKey(passcode: string) {
 }
 
 /**
+ * @deprecated Use deleteDevice with no arguments to get the same behavior.
  * Clears local device keys from the current browser instance. This will require the user to enter their passcode the next time they want to use this browser on this machine.
  * This method should usually be called whenever the current user logs out of your application.
  */
-export function deauthorizeDevice() {
+export const deauthorizeDevice = () => deleteDevice().then((deletedDevice) => ({transformKeyDeleted: deletedDevice > 0}));
+
+/**
+ * Deletes a device. If deleting the current device, the user will have to enter their passcode the next time they want to use this browser on this machine.
+ * This method should usually be called whenever the current user logs out of your application or you're aware a device of their's shouldn't have access.
+ * @param {number | undefined} deviceId The device id to delete. If undefined, the current device will be deleted and local storage will be cleared.
+ */
+// TODO: detect if deviceId is the same as the currently operating device and set deletingCurrentDevice to true
+export const deleteDevice = (deviceId?: number) => {
     checkSDKInitialized();
-    const payload: MT.DeauthorizeDevice = {
-        type: "DEAUTHORIZE_DEVICE",
-        message: null,
+    const deletingCurrentDevice = deviceId === undefined;
+    const payload: MT.DeleteDevice = {
+        type: "DELETE_DEVICE",
+        message: deletingCurrentDevice ? undefined : deviceId,
```

Comment:
I think this is simpler?
```suggestion
        message: deviceId,
```

Reply:
It is if we can't come up with a solution to the TODO.